### PR TITLE
Use FileReader in macOS native apps using WebKit

### LINKED ISSUE #509: Cannot save files on Outlook Mac
FileSaver works well on all browsers including Safari, but shows this popup in Outlook for Mac:

31315907-fef5c60c-ac2a-11e7-996c-a64bf267a357

OS: MacOS High Sierra 
ENV: Outlook Client 15.41 and 16.20 (v16 doesn't show popups at all)

Safari agent where it is working well:
`5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.0.2 Safari/605.1.15`

Outlook internal Safari agent where it is failing:
`5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)`

### PR DESCRIPTION
**Pull request by rajivshah3**
In a macOS native app that uses WebKit, calling `saveAs` on a `Blob` attempts to open the blob URL. This results in a message that says `There is no application set to open the URL blob:http...` (see https://github.com/iotaledger/spark-wallet/issues/89#issuecomment-568169805). 

As in Safari, the `FileReader` API should be used. However, changes need to be made to the browser detection because of a few differences between Safari and WebKit:

- `HTMLAnchorElement.download` is defined
- `typeof FileReader` returns `'function'` instead of `'object'`

With these changes, the expected behavior is achieved:

image

Fixes #509

### CODE REVIEW

**Comment by Floriferous** [comment]
I have a similar issue where `HTMLAnchorElement` itself is undefined in a node test environment, could we also first check if it exists before trying to access its prototype?

Like this:
`HTMLAnchorElement && !isWebKit && ('download' in HTMLAnchorElement.prototype)`

I suggest the isWebkit check is done before trying to find 'download', as isWebKit is a boolean that has already been calculated, so no need to perform the `in` operation if it is not needed.

**Comment by jdeanquin-dg** [comment]
> I have a similar issue where `HTMLAnchorElement` itself is undefined in a node test environment, could we also first check if it exists before trying to access its prototype?
> 
> Like this:
> `HTMLAnchorElement && !isWebKit && ('download' in HTMLAnchorElement.prototype)`
> 
> I suggest the isWebkit check is done before trying to find 'download', as isWebKit is a boolean that has already been calculated, so no need to perform the `in` operation if it is not needed.

Hello @Floriferous  I have submitted a PR to fix the problem in node environments since my tests were failing even if not trying to test FileSaver at all #620 620

**Comment by eligrey** [comment]
lgtm 👍

**Comment by eligrey** [comment]
LGTM; I will merge your changes once you bump the version

File: src/FileSaver.js [hunk 1]
```diff
@@ -137,7 +140,7 @@ var saveAs = _global.saveAs || (
     var isSafari = /constructor/i.test(_global.HTMLElement) || _global.safari
     var isChromeIOS = /CriOS\/[\d]+/.test(navigator.userAgent)
 
-    if ((isChromeIOS || (force && isSafari)) && typeof FileReader !== 'undefined') {
+    if (((isChromeIOS || (force && isSafari)) && typeof FileReader === 'object') || isWebKit && typeof FileReader === 'function') {
```
Comment (rajivshah3) [inline src/FileSaver.js]: Just realized this would essentially revert https://github.com/eligrey/FileSaver.js/pull/577, will fix

**Comment by rajivshah3** [comment]
@eligrey will do. Do you want me to rebuild the dist as well?

**Comment by eligrey** [comment]
Yeah, you should also re-build dist. Thanks for your PR!

**Comment by rajivshah3** [comment]
@eligrey just tested, still works on iOS Chrome

**Pull request merged**